File: app/src/main/java/com/monti/kristo/montikristo/rest/apiclient.java
package com.monti.kristo.montikristo.rest;


import com.monti.kristo.montikristo.rest.network.ApiInterface;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class apiclient {

    public static final String BASE_URL = "https://montikristo.com/api/";
    public static final String BASE_URL_LOCAL = "http://192.168.1.11:3000/api/";

    private static Retrofit retrofit = null;
    private static apiclient apiClientInstance;

    private apiclient() {
        retrofit = new Retrofit.Builder()
                .baseUrl(BASE_URL)
                .addConverterFactory(GsonConverterFactory.create())
                .build();
    }

    public static synchronized apiclient getApiClientInstance() {

        if (apiClientInstance == null) {
            apiClientInstance = new apiclient();
        }

        return apiClientInstance;
    }

    public ApiInterface getApi() {
        return retrofit.create(ApiInterface.class);
    }
}

File: app/src/main/java/com/monti/kristo/montikristo/rest/apiclient.kt
package com.monti.kristo.montikristo.rest


import com.monti.kristo.montikristo.rest.network.ApiInterface
import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor

import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

class apiclient private constructor() {

    val api: ApiInterface
        get() = retrofit!!.create(ApiInterface::class.java)

    init {
        retrofit = Retrofit.Builder()
                .baseUrl(BASE_URL_LOCAL)
                .client(OkHttpClient().newBuilder().addInterceptor(HttpLoggingInterceptor().setLevel(HttpLoggingInterceptor.Level.BODY)).build())
                .addConverterFactory(GsonConverterFactory.create())
                .build()
    }

    companion object {

        val BASE_URL = "https://montikristo.com/api/"
        val BASE_URL_LOCAL = "http://192.168.1.22:3000/api/"

        private var retrofit: Retrofit? = null
        private var apiClientInstance: apiclient? = null

        @Synchronized
        fun getApiClientInstance(): apiclient {

            if (apiClientInstance == null) {
                apiClientInstance = apiclient()
            }

            return apiClientInstance as apiclient
        }
    }
}
